contents/blog/mobile-session-replay.md

```diff
@@ -0,0 +1,139 @@
+---
+title: How we built mobile replay (and why it took so long)
+date: 2024-10-08
+author:
+ - ian-vanagas
+ - manoel-aranda-neto
+rootpage: /blog
+featuredImage: >- 
+  https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/posthog-engineering-blog.png
+featuredImageType: full
+tags:
+ - Guides
+---
+
+[Session replay](/session-replay) is one of the most powerful tools for understanding user behavior. Web session replay has been a core part of PostHog for a long time now (it was built in our first hackathon), but mobile teams have had to wait longer.
+
+Finally, it's here on [iOS](/docs/session-replay/ios), [Android](/docs/session-replay/android), and [React Native](/docs/session-replay/react-native) (with [Flutter](https://github.com/PostHog/posthog-flutter/issues/69) coming soon). 
+
+What took so long? Recording them in mobile apps is much trickier than in web apps. This post goes over why and how we finally managed to make it happen.
```

Comment:
```suggestion
Web session replay has been a core part of PostHog since our first hackathon way back in [insert year/date + link about it if we have one], but we only released our first betas for mobile apps in [xxx] this year.

Why the long wait? 

1. It made sense for us to focus mainly on web-based B2B customers during our early-stage phase.

2. We shipped a whole suite of products, including feature flags, experiments, surveys and (more recently) our own data warehouse.

3. Building reliable and useful replay for mobile apps is way harder than web.

Happily, after several months of hard work from Manoel and the replay team, we have free-to-use betas for [iOS](/docs/session-replay/ios), [Android](/docs/session-replay/android), and [React Native](/docs/session-replay/react-native), with [Flutter](https://github.com/PostHog/posthog-flutter/issues/69) coming soon. 

This post covers how it works and some the technical challenges we overcame it make it.
```

Bit of a rework:
- Didn't love the opening line, which felt a bit puffy for an engineering blog.
- Please fill in the blanks, but there was a lack of context around timescales.
- Adding more context for the wait beyond "it's hard". There were business reasons, too.